lib/main.js

```diff
@@ -841,6 +843,17 @@ exports.typeset = function (data,callback) {
   if (serverState == STATE.READY) StartQueue();
 }
 
+// main API, callback and promise compatible
+exports.typeset = function (data, callback) {
+    if (callback) cbTypeset(data, callback);
+    else return new Promise(function (resolve, reject) {
+        cbTypeset(data, function (output, input) {
+            if (output.errors) reject(output.errors);
```

Comment:
The problem turns out to be that the CHTML undefined character signals are not being trapped — only the SVG ones.  So your first `typeset()` call shows them because it has `svg: true`, but the second doesn't, because it is only for html.

You can fix this by adding

```
      MathJax.Hub.Register.MessageHook("CommonHTML Jax - unknown char",function (message) {
        AddError("CHTML - Unknown character: U+"+message[1].toString(16).toUpperCase()+
                    " in "+(message[2].fonts||["unknown"]).join(","),!undefinedChar);
      });
```

just after the corresponding SVG version at around line 198.